library/src/main/java/org/apache/apex/malhar/lib/state/managed/AbstractManagedStateImpl.java

```diff
@@ -176,6 +177,12 @@ public void setup(OperatorContext context)
     operatorContext = context;
     fileAccess.init();
 
+    if ( timeBucketAssigner == null ) {
+      // set default time bucket assigner
+      MovingBoundaryTimeBucketAssigner movingBoundaryTimeBucketAssigner = new MovingBoundaryTimeBucketAssigner();
+      movingBoundaryTimeBucketAssigner.setPurgeListener(this);
```

Comment:
*.setPurgeListener(this) is not needed here. This anyway setting after the if block.